### ISSUE TITLE
Transfer Errors, No Hatch, Item Options not shown

### ISSUE THREAD

**Issue by MaximPro** [finndev/PokeBuddy#48]
Hello,

Thanks for developing such a great bot.
You did a great job so far. Many people love it around the world!

I'm using it now for several days. I also see the updates coming in daily - amazing.
Also I'm representing a community with lots of members and their issues. (100000+)

**We found the following, that needs fixes/improvements:**

- It randomly stucks more often UI-Updating to long and only stop and start lets it go to work again.

- Reconnect Feature, after mostly around 1-2 hours it just can't connect anymore, except manual stop and restart, wich works really nice. (should use a random timer between 1-5 minutes to reconnect automatically - maybe if certain log statements appear to often or UI is updateting to long)

- The Pokemon Options don't work - you can just choose 1 pokemon for transfer and evolve? And it doesn't even transfers those.

- It doesn't show anything in Item-Options, there should be options how much items to recycle, keep and if it should use lucky eggs, lure etc.

- I doesn't seem to hatch eggs or have options for that in the GUI (in the Config File it is, but doesn't seem to work) - nothing in the log about that.

- Use the highest avaiable CP Pokemon to evolve. (sometimes it uses the lowest CP)
- FIlter for evolve, like only evolve above 600 CP or don't evolve lowest evolution.

**Feature Requests:**

- There should be an option to transfer or evolve pokemons manualy throught the list on the right. (both should have an dialog yes/no for security reasons)

- Much needed is an CP-Limit for pokemon to send professor

- Also if a specific pokemon is already on the pokedex and is really low, it should be send away (use a better, editable transfer list for that) - (there are problems with full 250 pokemon bags)

- Waypoints to go
- area to stay in
- slowly walk without waypoints
- avoid water

Bonus:

- build a big donate button into the gui ;)  (top-center)

Probably there is more.
I added a small video about the pokemon and item settings issue...

However You guys do an amazing job!
Please use this list to GO for it all!

Thanks
prog4mer and the Community

**Comment by joaopedrosantos** [comment]
Does it hatch eggs? I don't think so

If not, thats a suggestion :)

**Comment by joaopedrosantos** [comment]
Allow checking the stats of each Pokémon - candy, IV, etc

probably a Menu UI for the pokemons

**Comment by wilbak** [comment]
I second

**Comment by brijei** [comment]
I does hatch eggs. only you need to put the eggs manually in their incubators.

**Comment by MaximPro** [comment]
Yes, thats the Point of having this bot. To automate everything.

Go for it, post things you like 😉

**Comment by curtiola** [comment]
Good post, I like the suggestions :)

**Comment by BlueSerenity** [comment]
Please extend the radius on finding and catching pokemon, its way too small. The radius on when it goes to a nearby pokestop is like 10x larger..

**Comment by TFSho** [comment]
@BlueSerenity 

When task "Catch Pokemon" is the only thing running, are you moving or not? If not at all, what you are probably seeing is the side effect of Niantic's "bumdate". Since we can no longer query their servers for the coordinates of pokemon, PB doesn't know where to go. It will only catch pokemon you are on top of. Running Farm and Catch seems to be the only workaround for moving and catching.

**Comment by brijei** [comment]
the evolve function doesn't work
